_posts/2018-10-09-how-we-work-pair-programming.md

```diff
@@ -0,0 +1,110 @@
+each role, with the caveat that we're still figuring it out ourselves.
+
+### Driver/Navigator
+
+The Driver is the one who operates the keyboard and mouse. They should think out loud and voice
+their expectations so that the Navigator knows what's going on.
+
+The Navigator tends to do most everything else:
+
+- Note opportunities for refactoring
+- Keep scope of story small
+- Ask for breaks on a regular basis
+- Encourage the Driver to commit frequently
+- Follow the storyline and stop the driver if it doesn’t make sense
+- Actively confirm or reject assumptions
+- Ask for arguments if a driver takes a non obvious solution: "Why did you do it that way?”
+- Look for the nearest exit: “If you fix that test now we can do a commit first, the build will be
+  green”
+
+## Logistics
+
+The logistics of how we pair is importannt because it increases our effectiveness. We almost always
+drive on our own machine, but there are plenty of times when we drive on someone else's machine with
+their editor so we can be more editor-agnostic.
+
+### In-Person Pairing
+
+In the past we paired our desks such that each desk had a monitor that mirrored what the other desk
+was doing. This allowed us to stay at our own desk but still effectively pair. But we appreciate
+being shoulder-to-shoulder more than remaining in the comfort of our own desk, and so have adopted
+another setup.
+
+Our (currently) prefered setup, which you can see an example of in the photo at the top of this
+post, is to pair shoulder-to-shoulder with 2 mirrored monitors, 2 keyboards (only 1 active at a time
+so as not to confuse roles), and 2 mice (both active so that the Navigator can point to things on
+the screen). We have multiple desks that are designated as pairing stations, and while each of us
+have "our own desk", we're usually working at one of the pairing stations.
+
+### Remote Pairing
+
+We have 1 remote engineer, plus we contract with a small company to help us improve our skills and
+they are all remote. So at any given time we may have 3-4 people on our team who are remote. While
+we think in-person pairing is much better than remote pairing, we have experimented with a lot of
+tools to make remote pairing better.
+
+We use Slack for a lot of our communication and it works well enough for video calls and screen
+sharing. However, if we're having video issues we switch to Skype because it seems to have much
+clearer video even on slower connections, and while Skype doesn't allow us to draw on each other's
+screen or control their machine, that doesn't hinder us too much. Instead it improves our
+communication skills because we can no longer circle the thing we're talking about, we have to use
+words :wink:
+
+For those of us who use terminal-based editors such as emacs and vim, we use [tmate][] and [SSH][],
+along with an audio call using Slack or whatever the pair feels like: Hangouts, Skype, Riot,
+old-fashioned telephone call, etc.
+
+### Promiscuous Pairing
```

Comment:
If we're looking to cut stuff out, I think we could do without this teaser for promiscuous pairing.